diff --git a/src/GUI/AddOnsController.cxx b/src/GUI/AddOnsController.cxx
index 39e7e44bd..40a4ce421 100644
--- a/src/GUI/AddOnsController.cxx
+++ b/src/GUI/AddOnsController.cxx
@@ -215,7 +215,7 @@ QString AddOnsController::addSceneryPath() const
     SGPath p(path.toStdString());
     bool isValid = false;
 
-    for (const auto& dir: {"Objects", "Terrain", "Buildings", "Roads", "Pylons", "NavData", "Airports"}) {
+    for (const auto& dir: {"Objects", "Terrain", "Buildings", "Roads", "Pylons", "NavData", "Airports", "Orthophotos"}) {
         if ((p / dir).exists()) {
             isValid = true;
             break;
@@ -228,7 +228,7 @@ QString AddOnsController::addSceneryPath() const
         mb.setStandardButtons(QMessageBox::Yes | QMessageBox::No);
         mb.setDefaultButton(QMessageBox::No);
         mb.setInformativeText(tr("Added scenery should contain at least one of the following "
-                                 "folders: Objects, Terrain, Buildings, Roads, Pylons, NavData, Airports."));
+                                 "folders: Objects, Terrain, Buildings, Roads, Pylons, NavData, Airports, Orthophotos."));
         mb.exec();
 
         if (mb.result() == QMessageBox::No) {
diff --git a/src/Scenery/tileentry.cxx b/src/Scenery/tileentry.cxx
index 31d14c713..04cddac30 100644
--- a/src/Scenery/tileentry.cxx
+++ b/src/Scenery/tileentry.cxx
@@ -46,6 +46,14 @@ TileEntry::TileEntry ( const SGBucket& b )
       _current_view(false),
       _time_expired(-1.0)
 {
+    bool use_photoscenery = fgGetBool("/sim/rendering/photoscenery/enabled");
+    if (use_photoscenery) {
+        _orthophoto = simgear::Orthophoto::fromBucket(tile_bucket, globals->get_fg_scenery());
+        if (_orthophoto) {
+            simgear::OrthophotoManager::instance()->registerOrthophoto(tile_bucket.gen_index(), _orthophoto);
+        }
+    }
+    
     tileFileName += ".stg";
     _node->setName(tileFileName);
     // Give a default LOD range so that traversals that traverse
@@ -62,6 +70,14 @@ TileEntry::TileEntry( const TileEntry& t )
   _current_view(t._current_view),
   _time_expired(t._time_expired)
 {
+    bool use_photoscenery = fgGetBool("/sim/rendering/photoscenery/enabled");
+    if (use_photoscenery) {
+        _orthophoto = simgear::Orthophoto::fromBucket(tile_bucket, globals->get_fg_scenery());
+        if (_orthophoto) {
+            simgear::OrthophotoManager::instance()->registerOrthophoto(tile_bucket.gen_index(), _orthophoto);
+        }
+    }
+
     _node->setName(tileFileName);
     // Give a default LOD range so that traversals that traverse
     // active children (like the groundcache lookup) will work before
@@ -115,5 +131,9 @@ TileEntry::removeFromSceneGraph()
             parent->removeChild( _node.get() );
         }
     }
+
+    if (_orthophoto) {
+        simgear::OrthophotoManager::instance()->unregisterOrthophoto(tile_bucket.gen_index());
+    }
 }
 
diff --git a/src/Scenery/tileentry.hxx b/src/Scenery/tileentry.hxx
index 56f2550cd..345e31af3 100644
--- a/src/Scenery/tileentry.hxx
+++ b/src/Scenery/tileentry.hxx
@@ -29,6 +29,9 @@
 # error This library requires C++
 #endif
 
+#include <Main/globals.hxx>
+#include <Main/fg_props.hxx>
+
 #include <simgear/compiler.h>
 
 #include <vector>
@@ -36,6 +39,7 @@
 
 #include <simgear/bucket/newbucket.hxx>
 #include <simgear/misc/sg_path.hxx>
+#include <simgear/scene/util/OrthophotoManager.hxx>
 
 #include <osg/ref_ptr>
 #include <osgDB/ReaderWriter>
@@ -59,6 +63,8 @@ private:
     // Reference to DatabaseRequest object set and used by the
     // osgDB::DatabasePager.
     osg::ref_ptr<osg::Referenced> _databaseRequest;
+    // Overlay image/orthophoto for this tile
+    simgear::OrthophotoRef _orthophoto;
 
     /**
      * This value is used by the tile scheduler/loader to load tiles
